src/components/tree/tree.ts

```diff
@@ -0,0 +1,85 @@
+import { PluginObject } from 'vue';
+import Component from 'vue-class-component';
+import { Prop } from 'vue-property-decorator';
+
+import { ModulVue } from '../../utils/vue/vue';
+import { TREE_NAME } from '../component-names';
+import I18nPlugin from '../i18n/i18n';
+import TreeNodePlugin from '../tree-node/tree-node';
+import WithRender from './tree.html';
+
+export interface TreeNode<T> {
+    content: T;
+    childs?: TreeNode<T>[];
+}
+
+export enum MSelectOption {
+    NONE = '0',
+    SINGLE = '1',
+    MULTIPLE = '2'
+}
+
+@WithRender
+@Component
+export class MTree<T> extends ModulVue {
+
+    @Prop()
+    tree: TreeNode<T>[];
+
+    @Prop({
+        default: MSelectOption.NONE,
+        validator: value =>
+            value === MSelectOption.NONE ||
+            value === MSelectOption.SINGLE ||
+            value === MSelectOption.MULTIPLE
+    })
+    selectionNumber: MSelectOption;
+
+    @Prop({ default: 'information' })
+    selectionIcon: string;
+
+    @Prop()
+    externalSelectedNode: TreeNode<T>[];
+
+    @Prop({ default: false })
+    isAllOpen: boolean;
+
+    @Prop({ default: false })
+    isFileTree: boolean;
+
+    internalSelectedNode: TreeNode<T>[] = [];
+    emptyTreeTxt: string = this.$i18n.translate('m-tree:empty');
+
+    created(): void {
+        this.selectNode = this.externalSelectedNode ? this.externalSelectedNode : [];
+    }
+
+    isTreeEmpty(): boolean {
+        return !this.tree.length;
+    }
+
+    selectNewNode(node: TreeNode<T>): void {
+        this.selectNode = [node];
```

Comment:
Détail mais je pense que si this.selectNode représente le node présentement sélectionné la variable devrait s'appeller selectEDNode

Reply:
Corrigé